clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesDataProviderIntegrationTest.java

```diff
@@ -0,0 +1,1117 @@
+      new KubernetesV2CachingAgentDispatcher(objectMapper, registry);
+  private static final ImmutableList<KubernetesHandler> handlers =
+      ImmutableList.of(
+          new KubernetesDeploymentHandler(),
+          new KubernetesReplicaSetHandler(),
+          new KubernetesServiceHandler(),
+          new KubernetesPodHandler());
+  private static final KubernetesSpinnakerKindMap kindMap =
+      new KubernetesSpinnakerKindMap(handlers);
+  private static final GlobalResourcePropertyRegistry resourcePropertyRegistry =
+      new GlobalResourcePropertyRegistry(
+          handlers, new KubernetesUnregisteredCustomResourceHandler());
+  private static final AccountCredentialsRepository credentialsRepository =
+      new MapBackedAccountCredentialsRepository();
+  private static final KubernetesAccountResolver accountResolver =
+      new KubernetesAccountResolver(credentialsRepository, resourcePropertyRegistry);
+  private static final ProviderRegistry providerRegistry =
+      new DefaultProviderRegistry(
+          ImmutableList.of(kubernetesV2Provider), new InMemoryNamedCacheFactory());
+  private static final KubernetesCacheUtils cacheUtils =
+      new KubernetesCacheUtils(
+          providerRegistry.getProviderCache(kubernetesV2Provider.getProviderName()),
+          kindMap,
+          accountResolver);
+  private static final ImmutableSetMultimap<String, String> manifestsByNamespace =
+      ImmutableSetMultimap.<String, String>builder()
+          .putAll(
+              "backend-ns",
+              ImmutableSet.of(
+                  "backend-service.yml",
+                  "backend-rs-014.yml",
+                  "backend-pod-014.yml",
+                  "backend-rs-015.yml",
+                  "backend-pod-015.yml"))
+          .putAll(
+              "frontend-ns",
+              ImmutableSet.of(
+                  "frontend-service.yml",
+                  "frontend-deployment.yml",
+                  "frontend-rs-old.yml",
+                  "frontend-rs-new.yml",
+                  "frontend-pod-1.yml",
+                  "frontend-pod-2.yml"))
+          .build();
+
+  private static KubernetesV2ApplicationProvider applicationProvider =
+      new KubernetesV2ApplicationProvider(cacheUtils);
+  private static KubernetesV2ClusterProvider clusterProvider =
+      new KubernetesV2ClusterProvider(cacheUtils, kindMap);
+  private static KubernetesV2InstanceProvider instanceProvider =
+      new KubernetesV2InstanceProvider(cacheUtils, accountResolver);
+  private static KubernetesV2LoadBalancerProvider loadBalancerProvider =
+      new KubernetesV2LoadBalancerProvider(cacheUtils, kindMap);
+  private static KubernetesV2SearchProvider searchProvider =
+      new KubernetesV2SearchProvider(cacheUtils, kindMap, objectMapper, accountResolver);
+  private static KubernetesV2ServerGroupManagerProvider serverGroupManagerProvider =
+      new KubernetesV2ServerGroupManagerProvider(cacheUtils);
```

Comment:
Was about to ask if we're explicitly not asserting on entities that map to Spinnaker security groups, then remembered that Deck actually populates the non-details "Firewalls" view using the search endpoint rather than a /securityGroups endpoint. Also, given that this was broken forever and almost no one noticed, likely not worth adding any assertions on NetworkPolicies...

Reply:
The reason I didn't add tests on `KubernetesV2SecurityGroupProvider` is because as far as I can tell security groups are completely independent from the other types.  It probably would be good to add some tests on it but I think those can be a completely separate test that just deploys some `NetworkPolicies` alone and then makes some calls (to avoid adding more logic to this test).  My first round of refactoring doesn't affect security groups much which is why I didn't add the test now, but probably should in the near future.

Reply:
👍 The only reason I thought of it was I remembered that the `/clusters` endpoint, in addition to the strange behavior of returning services, also returns network policies.